docs/standard/native-interop/comwrappers-source-generation.md

```diff
@@ -0,0 +1,196 @@
+
+## Basic usage
+
+To use the COM interface generator, add the <xref:System.Runtime.InteropServices.Marshalling.GeneratedComInterfaceAttribute> and <xref:System.Runtime.InteropServices.GuidAttribute> attributes on the interface definition that you want to import from or expose to COM. The type must be marked `partial` and have `internal` or `public` visibility for the generated code to be able to access it.
+
+```csharp
+[GeneratedComInterface]
+[Guid("3faca0d2-e7f1-4e9c-82a6-404fd6e0aab8")]
+internal partial IFoo
+{
+    void Method(int i);
+}
+```
+
+Then, to expose a class that implements an interface to COM, add the <xref:System.Runtime.InteropServices.Marshalling.GeneratedComClassAttribute> to the implementing class. This class will also need to be `partial` and either `internal` or `public`.
+
+```csharp
+[GeneratedComClass]
+internal partial class Foo : IFoo
+{
+    public void Method(int i)
+    {
+        // Do things
+    }
+}
+```
+
+At compile time, the generator will create an implementation of the ComWrappers API, and you can use the <xref:System.Runtime.InteropServices.Marshalling.StrategyBasedComWrappers> type or a custom derived type to consume or expose the COM interface.
+
+```csharp
+[LibraryImport("MyComObjectProvider.dll")]
+static nint GetPointerToComInterface();
+
+[LibraryImport("MyComObjectProvider.dll")]
+static void GivePointerToComInterface(nint comObject);
+
+// Use the system to create a Runtime Callable Wrapper to use in managed code
+ComWrappers cw = new StrategyBasedComWrappers();
+nint ptr = GetPointerToComInterface();
+IFoo foo = (IFoo)cw.GetOrCreateObjectForComInterface(ptr);
+foo.Method(0);
+...
+// Use the system to create a COM Callable Wrapper to pass to unmanaged code
+ComWrappers cw = new StrategyBasedComWrappers();
+Foo foo = new();
+nint ptr = cw.GetOrCreateComInterfaceForObject(foo);
+GivePointerToComInterface(ptr);
+```
+
+## Customizing marshalling
+
+The COM interface generator respects the <xref:System.Runtime.InteropServices.MarshalAsAttribute> and <xref:System.Runtime.InteropServices.Marshalling.MarshalUsingAttribute> attributes to customize marshalling of parameters. For more information, see [Customize Parameter Marshalling](./customize-parameter-marshalling.md) If an interface has `string` parameters or return types, the <xref:System.Runtime.InteropServices.Marshalling.GeneratedComInterfaceAttribute.StringMarshalling?displayProperty=nameWithType> and <xref:System.Runtime.InteropServices.Marshalling.GeneratedComInterfaceAttribute.StringMarshallingCustomType?displayProperty=nameWithType> properties will apply to all parameters and return types of `string` in the interface if they don't have other marshalling attributes.
```

Comment:
> For more information, see [Customize Parameter Marshalling](./customize-parameter-marshalling.md)

That doc only covers `MarshalAs`? Would it make sense to also link to the tutorial for using custom marshallers with `MarshalUsing`? (And maybe add a note in the 'Customize Parameter Marshalling' to mention source generation and the associated customization method)

Reply:
Added notes in `Customize Parameter Marshalling` that MarshalAs is only partially supported with source generated interop, and linked to the page discussing MarshalUsing